### ISSUE TITLE
[cache] "CacheNotExistsException" seen when a new member joins a cluster where the primary member is creating and destroying jcaches

### ISSUE THREAD

**Issue by siwhiting** [hazelcast/hazelcast#11072]
I've encountered an issue with Hazelcast 3.8.x / 3.9 EA, with the following jcache exception being raised by a Hazelcast member joining the cluster:

```
Exception in thread "hz._hzInstance_1_dev.cached.thread-1" com.hazelcast.cache.CacheNotExistsException: Couldn't find cache config with name cache_142
	at com.hazelcast.cache.impl.AbstractCacheService.checkCacheSimpleConfig(AbstractCacheService.java:217)
	at com.hazelcast.cache.impl.AbstractCacheService.createDistributedObject(AbstractCacheService.java:197)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.doCreateProxy(ProxyRegistry.java:191)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.createProxy(ProxyRegistry.java:184)
	at com.hazelcast.spi.impl.proxyservice.impl.operations.PostJoinProxyOperation$1.run(PostJoinProxyOperation.java:63)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:228)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Aug 11, 2017 9:29:17 AM com.hazelcast.cache.impl.CacheService
```

I have 2 members configured in a cluster. 

Member 1 is in a loop
- Creates jcache (with unique name)
- sleep 1 sec
- Schedules destroy of cache 10 secs later
- sleep 1 sec

Member 2
- starts up
- sleeps for 10 secs
- shuts down

Occasionally member 2 will raise the "CacheNotExistsException" for one or more of the active jcaches created by Member 1 during cluster synchronization(?)

I've been running on Linux, and issue occurs on both: centos 6 and Ubuntu 16.4.

For reproduction, I've attached a test program (jcache-versions.tar.gz
[jcache-versions.tar.gz](https://github.com/hazelcast/hazelcast/files/1217312/jcache-versions.tar.gz)
) that should reproduce the issue

Reproduction steps

1. Unpack jcache-versions.tar.gz test program on two linux machines with Java 8.
2. Edit conf/cache_versions_hazelcast.xml on both machines, to configure host names/ip addresses for the cluster
3. On machine 1 - bin/jcache-versions.bash 
4. On machine 2 - while :; do bin/jcache-versions.bash 1 1; sleep 1; done

After a while, should see the issue on machine 2
e.g.

```
Starting Hazelcast connection
Aug 11, 2017 9:39:06 AM com.hazelcast.config.AbstractXmlConfigHelper
WARNING: Name of the hazelcast schema location incorrect using default
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
WARNING: [LOCAL] [dev] [3.9-EA] You configured your member address as host name. Please be aware of that your dns can be spoofed. Make sure that your dns configurations are correct.
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-EA] Resolving domain name 'sw-eng-app1' to address(es): [10.7.5.228]
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
WARNING: [LOCAL] [dev] [3.9-EA] You configured your member address as host name. Please be aware of that your dns can be spoofed. Make sure that your dns configurations are correct.
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-EA] Resolving domain name 'sw-eng-app2' to address(es): [10.7.5.117]
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-EA] Interfaces is disabled, trying to pick one address from TCP-IP config addresses: [sw-eng-app2/10.7.5.117, sw-eng-app1/10.7.5.228]
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-EA] Prefer IPv4 stack is true.
Aug 11, 2017 9:39:07 AM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-EA] Picked [sw-eng-app2]:5710, using socket ServerSocket[addr=/0:0:0:0:0:0:0:0,localport=5710], bind any local is true
Aug 11, 2017 9:39:07 AM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Hazelcast 3.9-EA (20170704 - f594093) starting at [sw-eng-app2]:5710
Aug 11, 2017 9:39:07 AM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
Aug 11, 2017 9:39:07 AM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Configured Hazelcast Serialization version: 1
Aug 11, 2017 9:39:07 AM com.hazelcast.spi.impl.operationservice.impl.BackpressureRegulator
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Backpressure is disabled
Aug 11, 2017 9:39:08 AM com.hazelcast.instance.Node
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Creating TcpIpJoiner
Aug 11, 2017 9:39:09 AM com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Starting 2 partition threads
Aug 11, 2017 9:39:09 AM com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Starting 3 generic threads (1 dedicated for priority tasks)
Aug 11, 2017 9:39:09 AM com.hazelcast.core.LifecycleService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] [sw-eng-app2]:5710 is STARTING
Aug 11, 2017 9:39:09 AM com.hazelcast.nio.tcp.TcpIpConnector
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Connecting to sw-eng-app1/10.7.5.228:5710, timeout: 0, bind-any: true
Aug 11, 2017 9:39:09 AM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Established socket connection between /10.7.5.117:55634 and sw-eng-app1/10.7.5.228:5710
Aug 11, 2017 9:39:15 AM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Cluster version set to 3.9
Aug 11, 2017 9:39:15 AM com.hazelcast.internal.cluster.ClusterService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] 

Members {size:2, ver:76} [
	Member [sw-eng-app1]:5710 - 67187ad3-86ca-4d05-8dab-3ac8e823c76d
	Member [sw-eng-app2]:5710 - f4c1900b-f6bc-4f89-b816-d6874ccfb4b9 this
]

Aug 11, 2017 9:39:15 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_440', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:15 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_439', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:15 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_438', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:15 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_437', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:15 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_436', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Exception in thread "hz._hzInstance_1_dev.cached.thread-2" com.hazelcast.cache.CacheNotExistsException: Couldn't find cache config with name cache_440
	at com.hazelcast.cache.impl.AbstractCacheService.checkCacheSimpleConfig(AbstractCacheService.java:217)
	at com.hazelcast.cache.impl.AbstractCacheService.createDistributedObject(AbstractCacheService.java:197)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.doCreateProxy(ProxyRegistry.java:191)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.createProxy(ProxyRegistry.java:184)
	at com.hazelcast.spi.impl.proxyservice.impl.operations.PostJoinProxyOperation$1.run(PostJoinProxyOperation.java:63)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:228)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Aug 11, 2017 9:39:16 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_441', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:17 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_436', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:18 AM com.hazelcast.core.LifecycleService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] [sw-eng-app2]:5710 is STARTED
Starting Destroy cache scheduler
Aug 11, 2017 9:39:18 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_442', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:19 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_437', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:20 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_443', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:21 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_438', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:22 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_444', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:23 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_439', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:24 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_445', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:25 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_440', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:26 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Added cache config: CacheConfig{name='cache_446', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 9:39:27 AM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-EA] Removed cache config: CacheConfig{name='cache_441', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Stopping Destroy cache scheduler
Stopping Hazelcast connection
```

**Comment by vbekiaris** [comment]
Hi @siwhiting , thanks for detailed issue report. This issue was resolved by https://github.com/hazelcast/hazelcast/pull/10902 which was merged after 3.9-EA was released. Can you please run again your test with members on the latest 3.9-SNAPSHOT? You will need this snapshot repos included in your build:
```
<repository>
    <id>sonatype-snapshots</id>
    <name>Sonatype Snapshot Repository</name>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <releases>
        <enabled>false</enabled>
    </releases>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>
```

If you can still reproduce the issue, please reopen the issue and we will investigate further.

**Comment by siwhiting** [comment]
The issue can be reproduced on 3.9-SNAPSHOT (though much less frequent). Can you please investigate further.

Thanks

Starting Hazelcast connection
Aug 11, 2017 3:38:26 PM com.hazelcast.config.AbstractXmlConfigHelper
WARNING: Name of the hazelcast schema location incorrect using default
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
WARNING: [LOCAL] [dev] [3.9-SNAPSHOT] You configured your member address as host name. Please be aware of that your dns can be spoofed. Make sure that your dns configurations are correct.
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-SNAPSHOT] Resolving domain name 'sw-eng-app1' to address(es): [10.7.5.228]
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
WARNING: [LOCAL] [dev] [3.9-SNAPSHOT] You configured your member address as host name. Please be aware of that your dns can be spoofed. Make sure that your dns configurations are correct.
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-SNAPSHOT] Resolving domain name 'sw-eng-app2' to address(es): [10.7.5.117]
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-SNAPSHOT] Interfaces is disabled, trying to pick one address from TCP-IP config addresses: [sw-eng-app2/10.7.5.117, sw-eng-app1/10.7.5.228]
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-SNAPSHOT] Prefer IPv4 stack is true.
Aug 11, 2017 3:38:27 PM com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [dev] [3.9-SNAPSHOT] Picked [sw-eng-app2]:5710, using socket ServerSocket[addr=/0:0:0:0:0:0:0:0,localport=5710], bind any local is true
Aug 11, 2017 3:38:27 PM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Hazelcast 3.9-SNAPSHOT (20170811 - fcc7787) starting at [sw-eng-app2]:5710
Aug 11, 2017 3:38:27 PM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
Aug 11, 2017 3:38:27 PM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Configured Hazelcast Serialization version: 1
Aug 11, 2017 3:38:27 PM com.hazelcast.spi.impl.operationservice.impl.BackpressureRegulator
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Backpressure is disabled
Aug 11, 2017 3:38:28 PM com.hazelcast.instance.Node
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Creating TcpIpJoiner
Aug 11, 2017 3:38:29 PM com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Starting 2 partition threads and 3 generic threads (1 dedicated for priority tasks)
Aug 11, 2017 3:38:29 PM com.hazelcast.internal.diagnostics.Diagnostics
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
Aug 11, 2017 3:38:29 PM com.hazelcast.core.LifecycleService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] [sw-eng-app2]:5710 is STARTING
Aug 11, 2017 3:38:29 PM com.hazelcast.nio.tcp.TcpIpConnector
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Connecting to sw-eng-app1/10.7.5.228:5710, timeout: 0, bind-any: true
Aug 11, 2017 3:38:29 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Established socket connection between /10.7.5.117:52460 and sw-eng-app1/10.7.5.228:5710
Aug 11, 2017 3:38:35 PM com.hazelcast.system
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Cluster version set to 3.9
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_48', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_47', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_46', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_51', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_50', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_49', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:35 PM com.hazelcast.internal.cluster.ClusterService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] 

Members {size:2, ver:10} [
	Member [sw-eng-app1]:5710 - b5dcd1b7-f068-49a6-8595-5b58bb100233
	Member [sw-eng-app2]:5710 - 2bc0dbc5-b6fd-4bf9-a01a-70489d9008c4 this
]

Aug 11, 2017 3:38:35 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Removed cache config: CacheConfig{name='cache_46', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Exception in thread "hz._hzInstance_1_dev.cached.thread-2" com.hazelcast.cache.CacheNotExistsException: Couldn't find cache config with name cache_46
	at com.hazelcast.cache.impl.AbstractCacheService.checkCacheSimpleConfig(AbstractCacheService.java:220)
	at com.hazelcast.cache.impl.AbstractCacheService.createDistributedObject(AbstractCacheService.java:200)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.doCreateProxy(ProxyRegistry.java:191)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.createProxy(ProxyRegistry.java:184)
	at com.hazelcast.spi.impl.proxyservice.impl.operations.PostJoinProxyOperation$1.run(PostJoinProxyOperation.java:63)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:228)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Aug 11, 2017 3:38:36 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_52', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:37 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Removed cache config: CacheConfig{name='cache_47', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:38 PM com.hazelcast.core.LifecycleService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] [sw-eng-app2]:5710 is STARTED
Starting Destroy cache scheduler
Aug 11, 2017 3:38:38 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_53', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:39 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Removed cache config: CacheConfig{name='cache_48', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:40 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_54', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:41 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Removed cache config: CacheConfig{name='cache_49', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:42 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_55', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:43 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Removed cache config: CacheConfig{name='cache_50', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:44 PM com.hazelcast.cache.impl.CacheService
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Added cache config: CacheConfig{name='cache_56', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=0, hotRestart=HotRestartConfig{enabled=false, fsync=false}}
Aug 11, 2017 3:38:45 PM com.hazelcast.instance.Node
INFO: [sw-eng-app2]:5710 [dev] [3.9-SNAPSHOT] Running shutdown hook... Current state: ACTIVE

**Comment by vbekiaris** [comment]
Thanks for the heads up, will look further into this.

**Comment by mdogan** [comment]
@vbekiaris: This problem happens when cache is destroyed while a member is joining. More specifically, it happens when destroy event is received after cache config is replicated during join (pre-join) but before cache proxy is created (post-join).

I think it's fine to suppress this exception during join in `PostJoinProxyOperation`. It's a safe scenario for jcache. Other data structures can create configs lazily, so they don't suffer from this problem.

**Comment by siwhiting** [comment]
Hi,

Thank you for the explanation of the issue being seen, and also 
confirming that the exception is benign in this scenario and can be ignored.

Suppressing the exception for the scenario seems sensible "fix".

Regards,
Simon

**Comment by mcculls** [comment]
Is there any chance this fix could be back-ported to 3.8.x?
